documentation/tech-directives/MatchMenteeAlgorithm.js

```diff
@@ -0,0 +1,139 @@
+      email: "shana.kroon@hotmail.de",
+      name: "Shana Kroon",
+      description: "Shana Kroon is an awesome person!",
+      rankedSkills: [
+        "Creativity",
+        "Interpersonal Communication",
+        "Critical Thinking",
+      ],
+      "time-zone": "UTC+5",
+      userType: "mentee",
+    },
+    mentee4: {
+      email: "leonora.reitz@gmail.com",
+      name: "Leonora Reitz",
+      description: "Leonora Reitz is an awesome person!",
+      rankedSkills: [
+        "Creativity",
+        "Interpersonal Communication",
+        "Critical Thinking",
+      ],
+      "time-zone": "UTC-2",
+      userType: "mentee",
+    },
+    mentee5: {
+      email: "mohammed.delany@yahoo.com",
+      name: "Mohammed Delany",
+      description: "Mohammed Delany is an awesome person!",
+      rankedSkills: [
+        "Creativity",
+        "Interpersonal Communication",
+        "Critical Thinking",
+      ],
+      "time-zone": "UTC+5",
+      userType: "mentee",
+    },
+  },
+};
+
+// algorithm
+let loggedmentor = {
+  Mentor1: {
+    email: "mohammed.delany@yahoo.com",
+    name: "Mohammed Delany",
+    description: "Mohammed Delany is an awesome person!",
+    rankedSkills: [
+      "Creativity",
+      "Interpersonal Communication",
+      "Critical Thinking",
+    ],
+    "time-zone": "UTC+5",
+    userType: "mentor",
+  },
+};
+
+// push mentor's skills to global variable
+var Mentorskills = loggedmentor.Mentor1.rankedSkills;
+
+// push mentor's timezone to global variable
+var s1 = loggedmentor.Mentor1["time-zone"];
+var mentorTZ = parseInt(s1.slice(3));
+// store menteeIDs and numbers of each mentee's overlapping skills
+// mentee1: [ 2, 10 ],
+// menteeUID: [<matched skills>, <diffTZ> ]
+let count = {};
+
+for (i in sampleData.users) {
+  // store number of matching skills
+  let num = 0;
+
+  // compare mentee's skills with mentor
+  // loop each mentee's skills
+  for (skill of sampleData["users"][i]["rankedSkills"]) {
+    // loop each mentorskills
+    for (h of Mentorskills) {
+      if (h == skill) {
+        // if match, num increment.
+        num++;
+      }
+    }
+  }
+  // store timezone as a string
+  var s2 = sampleData["users"][i]["time-zone"];
+  // convert string to integer
+  var menteeTZ = parseInt(s2.slice(3));
+  var diffTZ = menteeTZ + mentorTZ;
+  // add key of menteeID with value of number of overlapping skills, and bool of timezone
+  var arr = [num, diffTZ];
+  // store number of matching skills and timezone diff in count
+  count[i] = arr;
+}
+// if we want to display the two best mentor
+
+// display mentorUID and suggestedmentee.
+/* 
+For now, i set the algorithm to display Top 3 
+suggestedMentees according to number of matched skills 
+and closest timezone
+*/
+// sort the list of mentees
+// sort by num of skills
+var sortedByNum = Object.entries(count).sort((a, b) => {
+  if (a[0] < b[0]) return 1;
+  else {
+    return -1;
+  }
+});
+//print sorted mentees
+console.log(sortedByNum);ß
```

Comment:
I'm wondering if the symbol after the semi-colon (ß) is a typo, of if it's supposed to be there? It might have a purpose that I'm not aware of so I thought I'd ask

Reply:
It is a type. MB